generators/app/templates/src/app/screens/InitialLoading/index.js

```diff
@@ -0,0 +1,30 @@
+import { PureComponent } from 'react';
+import { compose } from 'recompose';
+import { connect } from 'react-redux';
+
+import Loadable from '../../components/Loadable';
+import Routes from '../../../constants/routes';
+
+class InitialLoading extends PureComponent {
+  static getDerivedStateFromProps(props) {
```

Comment:
Can't you use `componentDidUpdate` for this?
You're not updating the state here.

The only difference is that `componentDidUpdate` won't be called on the first render, so maybe you'll need to check for `initialLoading` in `componentDidMount` too.

Reply:
Although I'm not updating the state, this form seemed good to me since `getDerivedStateFromProps` always runs at the beginning of the life-cycle and while `props.initialLoading` is true it returns `null` so we avoid re-renders. Also, we avoid repeating code in `componentDidMount` and `componentDidUpdate`. What do you think? @guidoprinc @WFolini

Reply:
I'm just saying this isn't the best method to add this logic based in the documentation:

> getDerivedStateFromProps exists for only one purpose. It enables a component to update its internal state as the result of changes in props

>If you need to perform a side effect (for example, data fetching or an animation) in response to a change in props, use componentDidUpdate lifecycle instead.

(from: https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops and https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#when-to-use-derived-state).

If you'll need to repeat the same logic in both methods you can extract it to a function 😄.

Reply:
I see this fine! for me, there isn't a problem to use in that way

Reply:
@sfernandez11  and @WFolini, I support @pabloferro in this one. We should only use `getDerivedStateFromProps` functionality when we only need to change state with props values, and use `componentDidUpdate` as a replacement of `componentWillReceiveProps`